### PR TITLE
Fix calculation of server URL for JiraBuildAction

### PR DESCRIPTION
**Pull request by i386**
@johnou this contains the feedback you requested + fixed tests from #108 

Apologies for the breakages.

---
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/jira-plugin/115)

### CODE REVIEW

File: src/test/java/hudson/plugins/jira/UpdaterTest.java [hunk 1]
```diff
@@ -297,7 +297,7 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
         );
 
         // expected issue list
-        final List<JiraIssue> expectedIssuesToCarryOver = new ArrayList<JiraIssue>();
+        final Set<JiraIssue> expectedIssuesToCarryOver = Sets.newHashSet();
```
Comment (johnou) [inline src/test/java/hudson/plugins/jira/UpdaterTest.java]: Sets.newLinkedHashSet(); ? If running JDK 8+ order as stated in the javadocs is not predictable and can cause random test assertion failures.

Comment (johnou) [inline src/test/java/hudson/plugins/jira/UpdaterTest.java]: Might want to use it in other places too if ordering was predictable.

**Comment by johnou** [comment]
@i386 thanks!

**Comment by i386** [comment]
@johnou I went through and 's/HashSet/LinkedHashSet/' most of the tests. At least now it is consistent.

**Comment by johnou** [comment]
lgtm!

**Pull request merged**